### PR TITLE
LegacyStrings

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by tkelman** [comment]
UTF16String isn't going to work without LegacyStrings. Don't think that's how you should be calling transcode.

**Comment by musm** [comment]
fixed, I ran a couple of tests by adding packages and all seemed to work fine

File: src/WinRPM.jl [hunk 1]
```diff
@@ -345,34 +340,35 @@ deps(pkg::AbstractString, arch::AbstractString=OS_ARCH) = deps(select(lookup(pkg
     add = rpm_provides(rpm_requires(pkg))
     local packages::Vector{ParsedData}
     reqd = AbstractString[]
-    if isa(pkg,Packages)
+    if isa(pkg, Packages)
         packages = ParsedData[p for p in pkg.p]
     else
-        packages = ParsedData[pkg.pd,]
+        @show pkg.pd
```
Comment (tkelman) [inline src/WinRPM.jl]: debugging show shouldn't be left here

**Comment by musm** [comment]
fixed thoughts?

File: src/WinRPM.jl [hunk 2]
```diff
@@ -364,7 +360,7 @@ end
 
 install(pkg::AbstractString, arch::AbstractString=OS_ARCH; yes = false) = install(select(lookup(pkg, arch),pkg); yes = yes)
 
-function install(pkgs::Vector{Compat.ASCIIString}, arch = OS_ARCH; yes = false)
+function install(pkgs::Vector{AbstractString}, arch = OS_ARCH; yes = false)
```
Comment (tkelman) [inline src/WinRPM.jl]: you won't usually have a vector of abstractly typed elements

File: src/winrpm_bindeps.jl [hunk 3]
```diff
@@ -15,28 +15,29 @@ function package_available(p::RPM)
...
 end
 
 available_version(p::RPM) = lookup(p.package).p[1][xpath"version/@ver"][1]
-libdir(p::RPM,dep) = joinpath(dirname(dirname(@__FILE__)),"deps","usr","$(Sys.ARCH)-w64-mingw32","sys-root","mingw","bin")
+libdir(p::RPM,dep) = joinpath(dirname(dirname(@__FILE__)), "deps", "usr", "$(Sys.ARCH)-w64-mingw32", "sys-root", "mingw", "bin")
 pkg_name(p::RPM) = p.package
 
-provider(::Type{RPM},packages::Vector{Compat.ASCIIString}; opts...) = RPM(packages)
-provides(::Type{RPM},packages::AbstractArray, dep::LibraryDependency; opts...) =
+provider(::Type{RPM}, packages::Vector{AbstractString}; opts...) = RPM(packages)
```
Comment (tkelman) [inline src/winrpm_bindeps.jl]: same here, vector of abstractstring is not a commonly encountered type

**Comment by musm** [comment]
Why unnecessarily restrict function arguments?  I have been taught try to keep method signatures general since dispatch should take care of things. 

I can change it to `String` if you want, but isn't it better as follows

**Comment by tkelman** [comment]
vectors are not covariant

**Comment by musm** [comment]
fixed

File: src/winrpm_bindeps.jl (hunk 3 above)
Comment (musm) [inline src/winrpm_bindeps.jl]: Here's the thing though, the code already uses  vectors of abstractstring all over the place, see e.g. https://github.com/JuliaPackaging/WinRPM.jl/blob/master/src/WinRPM.jl#L406

Comment (tkelman) [inline src/winrpm_bindeps.jl]: sure, but if these are user facing then they need to handle types that will come from user input or code

**Pull request merged**